lib/Doctrine/DBAL/Platforms/OraclePlatform.php

```diff
@@ -608,6 +633,31 @@ private function getAutoincrementIdentifierName(Identifier $table)
     /**
      * {@inheritDoc}
      */
+    public function getListAllForeignKeysSQL(string $database): string
+    {
+        $databaseIdentifier = $this->normalizeIdentifier($database);
+        $quotedDatabaseIdentifier = $this->quoteStringLiteral($databaseIdentifier->getName());
+        return <<<SQL
+          SELECT cols.table_name,
+                 alc.constraint_name,
+                 alc.DELETE_RULE,
+                 cols.column_name "local_column",
+                 cols.position,
+                 r_cols.table_name "references_table",
+                 r_cols.column_name "foreign_column"
+            FROM all_cons_columns cols
+       LEFT JOIN all_constraints alc
+              ON alc.owner = cols.owner AND alc.constraint_name = cols.constraint_name
+       LEFT JOIN all_cons_columns r_cols
+              ON r_cols.owner = alc.r_owner AND r_cols.constraint_name = alc.r_constraint_name AND r_cols.position = cols.position
+           WHERE cols.owner = $quotedDatabaseIdentifier AND alc.constraint_type = 'R'
+           ORDER BY cols.table_name, cols.constraint_name, cols.position
```

Comment:
Please format all new SQL consistently (like this query, keyword on the left, one space, the rest on the righ).